# Prueba Técnica - Wise Athena
### Technical Test - Wise Athena

## Accesos
#### Access

In [1]:
import os
import sys

In [2]:
os.getcwd()

'c:\\Users\\maria\\OneDrive\\Escritorio\\The_Bridge\\Visual_Studio_Code\\Python\\Prueba_t-cnica_WA\\src'

In [3]:
rooth_path = os.path.dirname(os.getcwd())
rooth_path

'c:\\Users\\maria\\OneDrive\\Escritorio\\The_Bridge\\Visual_Studio_Code\\Python\\Prueba_t-cnica_WA'

In [4]:
sys.path.append(rooth_path)
sys.path = list(set(sys.path)) # Para evitar duplicados (To avoid duplicates)
sys.path

['',
 'c:\\Users\\maria\\.vscode\\extensions\\ms-toolsai.jupyter-2021.5.745244803\\pythonFiles\\lib\\python',
 'C:\\Program Files\\Python38\\python38.zip',
 'C:\\Program Files\\Python38\\lib\\site-packages\\Pythonwin',
 'C:\\Program Files\\Python38\\lib',
 'C:\\Program Files\\Python38\\lib\\site-packages\\win32',
 'C:\\Users\\maria\\.ipython',
 'C:\\Program Files\\Python38\\lib\\site-packages',
 'C:\\Program Files\\Python38\\DLLs',
 'c:\\Users\\maria\\.vscode\\extensions\\ms-toolsai.jupyter-2021.5.745244803\\pythonFiles',
 'c:\\Users\\maria\\OneDrive\\Escritorio\\The_Bridge\\Visual_Studio_Code\\Python\\Prueba_t-cnica_WA\\src',
 'c:\\Users\\maria\\OneDrive\\Escritorio\\The_Bridge\\Visual_Studio_Code\\Python\\Prueba_t-cnica_WA',
 'C:\\Program Files\\Python38',
 'C:\\Program Files\\Python38\\lib\\site-packages\\win32\\lib',
 'C:\\Program Files\\Python38\\lib\\site-packages\\IPython\\extensions']

## Librerías
### Libraries

In [5]:
import pandas as pd

from src.utils.mining_data_tb import changing_dates, split_column, check_dupli_nan

In [6]:
# from google.colab import drive
# drive.mount(‘/content/drive’)

## Lectura de csv
### Read csv

In [7]:
sellin = pd.read_csv("../data/sellin.csv", sep=",")
sellout_prov1 = pd.read_csv("../data/sellout-proveedor1.csv", sep=",")
sellout_prov2 = pd.read_csv("../data/sellout-proveedor2.csv", sep=",")
maestro_prod = pd.read_csv("../data/maestro_productos_prueba.csv", sep=",")
maestro_client = pd.read_excel("../data/maestro_clientes_prueba.xlsx",  engine='openpyxl')

## Entendiendo los datos
### Understanding Data

In [59]:
# Sellin: purchase from our client to supliers EX: Danone to Carrefour
sellin.head()

Sku       Date  Store_Id  Pieces  Price  Margin
0  4220015 2015-02-01       100    20.0  13.01    9.78
1  4220015 2015-02-01       116    20.0  13.01    9.78
2  4220015 2015-02-01       141    80.0  13.01    9.78
3  4220015 2015-02-01       146    10.0  13.01    9.78
4  4220015 2015-02-01       151    20.0  13.09    9.87

In [9]:
# Sellout_prov1: purchase from the first suplier to final client EX: Carrefour to final user
sellout_prov1.head()

Sku                         Date  Store_Id  Units  Price
0  4220015  2017-01-01 SO Diaria Piezas        18    6.0   15.6
1  4220015  2017-01-01 SO Diaria Piezas        24    0.0    0.0
2  4220015  2017-01-01 SO Diaria Piezas        39    0.0    0.0
3  4220015  2017-01-01 SO Diaria Piezas        44    0.0    0.0
4  4220015  2017-01-01 SO Diaria Piezas        46    0.0    0.0

In [60]:
# Sellout_prov2: purchase from the second suplier to final client EX: Mercadona to final user
sellout_prov2.head()

Sku       Date  Store_Id  Units  Price
0  4240039 2016-01-01      1000   23.0   16.0
1  4240039 2016-01-01      1001  116.0   16.0
2  4240039 2016-01-01      1002  146.0   16.0
3  4240039 2016-01-01      1003   17.0   16.8
4  4240039 2016-01-01      1004    5.0   16.7

In [11]:
# Type of produts of our client and competitors
maestro_prod.head()

Sku      Sku_Description   Sku_Category Sku_Subcategory
0  11058       Frosted Mango      Sugar-free     Cereal bars
1  29207  Nut Cheerios Mango   Super-flavour      Cereal box
2  38379    MultiGrain Mango      Sugar-free     Cereal bars
3  42671        Crunch Mango      Sugar-free     Cereal bars
4  54644          Corn Mango      Sugar-free     Cereal bars

In [12]:
# Shops and refer suppliers which sell their products
maestro_client.head()

NUM TDA              NOMBRE DE TDA       CADENA FORMATO  \
0        2  Bloody Sea Green BookShop  proveedor_1    Shop   
1        4  Funny Sky Blue GipsyTrade  proveedor_1    Shop   
2        5      Rare Green Laundrette  proveedor_1    Shop   
3        6        Angry Indigo Center  proveedor_1    Shop   
4        7           Tall Violet Deli  proveedor_1    Shop   

       MUNICIPIO/CIUDAD   ESTADO DIVISION DEPOSITO / DISTRIBUIDOR  \
0    Bizarre White Shop  Coalmar  SURESTE          Mogulval Corp.   
1    Bizarre White Shop  Coalmar  SURESTE          Mogulval Corp.   
2  Rare Orange Applepie  Coalmar  SURESTE        Eioval Aerospace   
3  Rare Orange Applepie  Coalmar  SURESTE        Eioval Aerospace   
4  Rare Orange Applepie  Coalmar  SURESTE        Eioval Aerospace   

       ESTRUCTURA GRADO  
0  DISTRIBUIDORES    VI  
1  DISTRIBUIDORES    VI  
2         FORANEO    VI  
3         FORANEO    VI  
4         FORANEO    VI

In [13]:
# Here we confirm we have only 2 different suppliers
maestro_client.CADENA.unique()

array(['proveedor_1', 'proveedor_2'], dtype=object)

Questions about:
- Which supliers sells more?
- Which product is the most sell? Which product minus?
- How is the benefit of our client in average?
- Which period is the most sell it?


## Bucamos Null y duplicados
### Checking for NaN and duplicates

In [17]:
check_dupli_nan("sellin", sellin)
check_dupli_nan("sellout_prov1", sellout_prov1)
check_dupli_nan("sellout_prov2", sellout_prov2)
check_dupli_nan("maestro_prod", maestro_prod)
check_dupli_nan("maestro_client", maestro_client)

sellin duplicates ----> True
sellin null ----> False
-------------------------- 

sellout_prov1 duplicates ----> False
sellout_prov1 null ----> False
-------------------------- 

sellout_prov2 duplicates ----> True
sellout_prov2 null ----> False
-------------------------- 

maestro_prod duplicates ----> False
maestro_prod null ----> False
-------------------------- 

maestro_client duplicates ----> False
maestro_client null ----> False
-------------------------- 



In [18]:
sellin = sellin.drop_duplicates()
check_dupli_nan("sellin", sellin)

sellout_prov2 = sellout_prov2.drop_duplicates()
check_dupli_nan("sellout_prov2", sellout_prov2)

sellin duplicates ----> False
sellin null ----> False
-------------------------- 

sellout_prov2 duplicates ----> False
sellout_prov2 null ----> False
-------------------------- 



## Sellin csv

In [19]:
sellin.head()

Sku        Date  Store_Id  Pieces  Price  Margin
0  4220015  02/01/2015       100    20.0  13.01    9.78
1  4220015  02/01/2015       116    20.0  13.01    9.78
2  4220015  02/01/2015       141    80.0  13.01    9.78
3  4220015  02/01/2015       146    10.0  13.01    9.78
4  4220015  02/01/2015       151    20.0  13.09    9.87

In [20]:
sellin.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4188765 entries, 0 to 4188782
Data columns (total 6 columns):
 #   Column    Dtype  
---  ------    -----  
 0   Sku       int64  
 1   Date      object 
 2   Store_Id  int64  
 3   Pieces    float64
 4   Price     float64
 5   Margin    float64
dtypes: float64(3), int64(2), object(1)
memory usage: 223.7+ MB


In [21]:
changing_dates(df=sellin, name_column="Date")

In [22]:
sellin.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4188765 entries, 0 to 4188782
Data columns (total 6 columns):
 #   Column    Dtype         
---  ------    -----         
 0   Sku       int64         
 1   Date      datetime64[ns]
 2   Store_Id  int64         
 3   Pieces    float64       
 4   Price     float64       
 5   Margin    float64       
dtypes: datetime64[ns](1), float64(3), int64(2)
memory usage: 223.7 MB


In [23]:
sellin.Pieces.unique()

array([ 20.  ,  80.  ,  10.  , ...,  31.75,  20.75, 137.5 ])

In [24]:
sellin.sort_values("Date")

Sku       Date  Store_Id  Pieces  Price  Margin
1239443  4361050 2015-01-04        18   15.12  12.99   10.67
3228928  4769520 2015-01-04      3372    4.80  35.54   27.38
3228927  4769520 2015-01-04      3344    7.20  35.54   27.38
3228926  4769520 2015-01-04      3323    7.20  35.54   27.38
3228925  4769520 2015-01-04      3318    4.80  35.54   27.38
...          ...        ...       ...     ...    ...     ...
774678   4250046 2017-12-05      1086   12.00  15.17   10.66
774679   4250046 2017-12-05      1108  108.00  15.17   10.66
774680   4250046 2017-12-05      1110   48.00  15.17   10.66
774682   4250046 2017-12-05      1150   12.00  15.17   10.66
2103459  4523924 2017-12-05      1119    4.80  43.41   29.84

[4188765 rows x 6 columns]

## Sellout_prov1

In [25]:
sellout_prov1.head()

Sku                         Date  Store_Id  Units  Price
0  4220015  2017-01-01 SO Diaria Piezas        18    6.0   15.6
1  4220015  2017-01-01 SO Diaria Piezas        24    0.0    0.0
2  4220015  2017-01-01 SO Diaria Piezas        39    0.0    0.0
3  4220015  2017-01-01 SO Diaria Piezas        44    0.0    0.0
4  4220015  2017-01-01 SO Diaria Piezas        46    0.0    0.0

In [26]:
sellout_prov1.Date.unique()

array(['2017-01-01 SO Diaria Piezas', '2017-01-02 SO Diaria Piezas',
       '2017-01-03 SO Diaria Piezas', '2017-01-04 SO Diaria Piezas',
       '2017-01-05 SO Diaria Piezas', '2017-01-06 SO Diaria Piezas',
       '2017-01-07 SO Diaria Piezas', '2017-01-08 SO Diaria Piezas',
       '2017-01-09 SO Diaria Piezas', '2017-01-10 SO Diaria Piezas',
       '2017-01-11 SO Diaria Piezas', '2017-01-12 SO Diaria Piezas',
       '2017-01-13 SO Diaria Piezas', '2017-01-14 SO Diaria Piezas',
       '2017-01-15 SO Diaria Piezas', '2017-01-16 SO Diaria Piezas',
       '2017-01-17 SO Diaria Piezas', '2017-01-18 SO Diaria Piezas',
       '2017-01-19 SO Diaria Piezas', '2017-01-20 SO Diaria Piezas',
       '2017-01-21 SO Diaria Piezas', '2017-01-22 SO Diaria Piezas',
       '2017-01-23 SO Diaria Piezas', '2017-01-24 SO Diaria Piezas',
       '2017-01-25 SO Diaria Piezas', '2017-01-26 SO Diaria Piezas',
       '2017-01-27 SO Diaria Piezas', '2017-01-28 SO Diaria Piezas',
       '2017-01-29 SO Diaria Pieza

In [27]:
split_column(df=sellout_prov1, column="Date", separation_object=" ", times=1)

In [28]:
sellout_prov1.Date.unique()

array(['2017-01-01', '2017-01-02', '2017-01-03', '2017-01-04',
       '2017-01-05', '2017-01-06', '2017-01-07', '2017-01-08',
       '2017-01-09', '2017-01-10', '2017-01-11', '2017-01-12',
       '2017-01-13', '2017-01-14', '2017-01-15', '2017-01-16',
       '2017-01-17', '2017-01-18', '2017-01-19', '2017-01-20',
       '2017-01-21', '2017-01-22', '2017-01-23', '2017-01-24',
       '2017-01-25', '2017-01-26', '2017-01-27', '2017-01-28',
       '2017-01-29', '2017-01-30', '2017-01-31', '2017-02-01',
       '2017-02-02', '2017-02-03', '2017-02-04', '2017-02-05',
       '2017-02-06', '2017-02-07', '2017-02-08', '2017-02-09',
       '2017-02-10', '2017-02-11', '2017-02-12', '2017-02-13',
       '2017-02-14', '2017-02-15', '2017-02-16', '2017-02-17',
       '2017-02-18', '2017-02-19', '2017-02-20', '2017-02-21',
       '2017-02-22', '2017-02-23', '2017-02-24', '2017-02-25',
       '2017-02-26', '2017-02-27', '2017-02-28', '2017-03-01',
       '2017-03-02', '2017-03-03', '2017-03-04', '2017-

In [29]:
changing_dates(df=sellout_prov1, name_column="Date")
sellout_prov1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1206365 entries, 0 to 1206364
Data columns (total 5 columns):
 #   Column    Non-Null Count    Dtype         
---  ------    --------------    -----         
 0   Sku       1206365 non-null  int64         
 1   Date      1206365 non-null  datetime64[ns]
 2   Store_Id  1206365 non-null  int64         
 3   Units     1206365 non-null  float64       
 4   Price     1206365 non-null  float64       
dtypes: datetime64[ns](1), float64(2), int64(2)
memory usage: 46.0 MB


## Sellout_prov2

In [30]:
sellout_prov2.head()

Sku        Date  Store_Id  Pieces   Price
0  4240039  2016/01/01      1000    23.0  $16.00
1  4240039  2016/01/01      1001   116.0  $16.00
2  4240039  2016/01/01      1002   146.0  $16.00
3  4240039  2016/01/01      1003    17.0  $16.80
4  4240039  2016/01/01      1004     5.0  $16.70

In [31]:
changing_dates(df=sellout_prov2, name_column="Date")
sellout_prov2.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4862026 entries, 0 to 4862028
Data columns (total 5 columns):
 #   Column    Dtype         
---  ------    -----         
 0   Sku       int64         
 1   Date      datetime64[ns]
 2   Store_Id  int64         
 3   Pieces    float64       
 4   Price     object        
dtypes: datetime64[ns](1), float64(1), int64(2), object(1)
memory usage: 222.6+ MB


In [32]:
sellout_prov2['Price'] = sellout_prov2['Price'].str.replace('$', '').astype(float)

In [33]:
sellout_prov2.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4862026 entries, 0 to 4862028
Data columns (total 5 columns):
 #   Column    Dtype         
---  ------    -----         
 0   Sku       int64         
 1   Date      datetime64[ns]
 2   Store_Id  int64         
 3   Pieces    float64       
 4   Price     float64       
dtypes: datetime64[ns](1), float64(2), int64(2)
memory usage: 222.6 MB


In [34]:
sellout_prov2 = sellout_prov2.rename(columns={'Pieces': 'Units'})

## Maestro_prod

In [35]:
maestro_prod.head()

Sku      Sku_Description   Sku_Category Sku_Subcategory
0  11058       Frosted Mango      Sugar-free     Cereal bars
1  29207  Nut Cheerios Mango   Super-flavour      Cereal box
2  38379    MultiGrain Mango      Sugar-free     Cereal bars
3  42671        Crunch Mango      Sugar-free     Cereal bars
4  54644          Corn Mango      Sugar-free     Cereal bars

In [36]:
maestro_prod.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1242 entries, 0 to 1241
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Sku              1242 non-null   int64 
 1   Sku_Description  1242 non-null   object
 2   Sku_Category     1242 non-null   object
 3   Sku_Subcategory  1242 non-null   object
dtypes: int64(1), object(3)
memory usage: 38.9+ KB


## Maestro_client

In [37]:
maestro_client.head()

NUM TDA              NOMBRE DE TDA       CADENA FORMATO  \
0        2  Bloody Sea Green BookShop  proveedor_1    Shop   
1        4  Funny Sky Blue GipsyTrade  proveedor_1    Shop   
2        5      Rare Green Laundrette  proveedor_1    Shop   
3        6        Angry Indigo Center  proveedor_1    Shop   
4        7           Tall Violet Deli  proveedor_1    Shop   

       MUNICIPIO/CIUDAD   ESTADO DIVISION DEPOSITO / DISTRIBUIDOR  \
0    Bizarre White Shop  Coalmar  SURESTE          Mogulval Corp.   
1    Bizarre White Shop  Coalmar  SURESTE          Mogulval Corp.   
2  Rare Orange Applepie  Coalmar  SURESTE        Eioval Aerospace   
3  Rare Orange Applepie  Coalmar  SURESTE        Eioval Aerospace   
4  Rare Orange Applepie  Coalmar  SURESTE        Eioval Aerospace   

       ESTRUCTURA GRADO  
0  DISTRIBUIDORES    VI  
1  DISTRIBUIDORES    VI  
2         FORANEO    VI  
3         FORANEO    VI  
4         FORANEO    VI

In [38]:
maestro_client.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2452 entries, 0 to 2451
Data columns (total 10 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   NUM TDA                  2452 non-null   int64 
 1   NOMBRE DE TDA            2452 non-null   object
 2   CADENA                   2452 non-null   object
 3   FORMATO                  2452 non-null   object
 4   MUNICIPIO/CIUDAD         2452 non-null   object
 5   ESTADO                   2452 non-null   object
 6   DIVISION                 2452 non-null   object
 7   DEPOSITO / DISTRIBUIDOR  2452 non-null   object
 8   ESTRUCTURA               2452 non-null   object
 9   GRADO                    2452 non-null   object
dtypes: int64(1), object(9)
memory usage: 191.7+ KB


In [39]:
maestro_client = maestro_client.rename(columns={'NUM TDA': 'Store_Id'})

## Merge DataFrames

First divide maestro client in 2 dataframes: Supplier 1 and Supplier 2

In [40]:
maestro_client_prov1 = maestro_client.loc[maestro_client["CADENA"] == "proveedor_1"]
maestro_client_prov2 = maestro_client.loc[maestro_client["CADENA"] == "proveedor_2"]

Merge sellout_prov1 with maestro_client_prov1 

In [41]:
# First supplier
df_merge1 = sellout_prov1.merge(maestro_client_prov1)

In [42]:
# Second supplier
df_merge2 = sellout_prov2.merge(maestro_client_prov2)

Comparing the difference of rows to see if match every item

In [43]:
# The result is a large number of no matches for first supplier
sellout_prov1.shape[0] - df_merge1.shape[0]

28985

In [44]:
# Here we can see that we have some shops that didn't buy anything for second supplier
sellout_prov2.shape[0] - df_merge2.shape[0]

-323

Then I compare if all the store_Id which have in products chart is in the list of stores chart

In [45]:
# Here pass the array of store_id to list
lista_prov1 = list(sellout_prov1.Store_Id.unique())
lista_prov2 = list(sellout_prov2.Store_Id.unique())

lista_maestro_prov1 = list(maestro_client_prov1.Store_Id.unique())
lista_maestro_prov2 = list(maestro_client_prov2.Store_Id.unique())

In [46]:
# Here I compare if the store id of products chart is in stores chart 
for x in lista_prov1:
    if x in lista_maestro_prov1:
        continue
    else:
        print(x)

174
712
681
78
94
301
675
676
709
711


In [47]:
# Again with second supplier
for x in lista_prov2:
    if x in lista_maestro_prov2:
        continue
    else:
        print(x)

So in this case we have two options:
    <br />1. Talk with the client and tell the problem to know which stores are these
    <br />2. Continue ignoring these stores

After this I checked if in store's chart has any Id duplicated

In [48]:
#In the case of the first supplier we have two store with the same Id
maestro_client_prov1[maestro_client_prov1.duplicated("Store_Id")]

Store_Id          NOMBRE DE TDA       CADENA FORMATO MUNICIPIO/CIUDAD  \
235       143  Empty Turquoise Baker  proveedor_1    Shop  Clean Aqua Soil   

       ESTADO   DIVISION        DEPOSITO / DISTRIBUIDOR ESTRUCTURA GRADO  
235  Nabargus  OCCIDENTE  Colorvit Ifonforge Industries    FORANEO   III

In [49]:
maestro_client_prov1.loc[(maestro_client_prov1["Store_Id"] == 143),:]

Store_Id          NOMBRE DE TDA       CADENA FORMATO MUNICIPIO/CIUDAD  \
115       143  Empty Turquoise Baker  proveedor_1    Shop  Clean Aqua Soil   
235       143  Empty Turquoise Baker  proveedor_1    Shop  Clean Aqua Soil   

       ESTADO   DIVISION        DEPOSITO / DISTRIBUIDOR      ESTRUCTURA GRADO  
115   Egusgul   NOROESTE          Gornvalval Industrail  DISTRIBUIDORES    IV  
235  Nabargus  OCCIDENTE  Colorvit Ifonforge Industries         FORANEO   III

In [50]:
# Again for the second supplier
maestro_client_prov2[maestro_client_prov2.duplicated("Store_Id")]

Store_Id            NOMBRE DE TDA       CADENA  FORMATO  \
2422      2568  Impressive Coral BigCom  proveedor_2  Grocery   

          MUNICIPIO/CIUDAD    ESTADO   DIVISION DEPOSITO / DISTRIBUIDOR  \
2422  Legendary Azure Tree  Mobartho  PENÍNSULA      Valarar Toughworks   

     ESTRUCTURA GRADO  
2422    FORANEO    VI

In [51]:
maestro_client_prov2.loc[(maestro_client_prov2["Store_Id"] == 2568),:]

Store_Id            NOMBRE DE TDA       CADENA  FORMATO  \
2409      2568  Impressive Coral BigCom  proveedor_2  Grocery   
2422      2568  Impressive Coral BigCom  proveedor_2  Grocery   

          MUNICIPIO/CIUDAD    ESTADO   DIVISION DEPOSITO / DISTRIBUIDOR  \
2409    First Yellow Tower  Mobartho  PENÍNSULA      Valarar Toughworks   
2422  Legendary Azure Tree  Mobartho  PENÍNSULA      Valarar Toughworks   

     ESTRUCTURA GRADO  
2409    FORANEO    VI  
2422    FORANEO    VI

Same Id different shops, again we have two options:
<br />1. Change the Store_Id for the correct Id, asking the client.
<br />2. Continue deleting the second one (235 in the index first supplier) (2422 in the index second supplier)

So now I concat the dataframes ignoring the problems

In [52]:
df_merge1 = sellout_prov1.merge(maestro_client_prov1)
df_merge2 = sellout_prov2.merge(maestro_client_prov2)

df_finalmerge = pd.concat([df_merge1, df_merge2])
df_finalmerge

Sku       Date  Store_Id  Units  Price           NOMBRE DE TDA  \
0        4220015 2017-01-01        18   6.00   15.6    Singing Black Center   
1        4220015 2017-01-02        18  15.00   15.6    Singing Black Center   
2        4220015 2017-01-03        18  10.00   15.6    Singing Black Center   
3        4220015 2017-01-04        18   8.00   15.6    Singing Black Center   
4        4220015 2017-01-05        18  12.00   15.6    Singing Black Center   
...          ...        ...       ...    ...    ...                     ...   
4862344  4625287 2016-08-20      3537   0.50    5.9  Singing Golden Florist   
4862345  4625287 2016-08-21      3537   1.50    5.9  Singing Golden Florist   
4862346  4625287 2016-08-22      3537   0.50    5.9  Singing Golden Florist   
4862347  4625287 2016-08-26      3537   0.50    8.9  Singing Golden Florist   
4862348  4625287 2016-08-28      3537   0.25    8.9  Singing Golden Florist   

              CADENA   FORMATO           MUNICIPIO/CIUDAD     ESTADO  \
0        proveedor_1      Shop           Evil White Baton   Alloralf   
1        proveedor_1      Shop           Evil White Baton   Alloralf   
2        proveedor_1      Shop           Evil White Baton   Alloralf   
3        proveedor_1      Shop           Evil White Baton   Alloralf   
4        proveedor_1      Shop           Evil White Baton   Alloralf   
...              ...       ...                        ...        ...   
4862344  proveedor_2  AllInOne  Enlightened Fuchsia Comet  Barthovit   
4862345  proveedor_2  AllInOne  Enlightened Fuchsia Comet  Barthovit   
4862346  proveedor_2  AllInOne  Enlightened Fuchsia Comet  Barthovit   
4862347  proveedor_2  AllInOne  Enlightened Fuchsia Comet  Barthovit   
4862348  proveedor_2  AllInOne  Enlightened Fuchsia Comet  Barthovit   

            DIVISION DEPOSITO / DISTRIBUIDOR ESTRUCTURA GRADO  
0        METRO-NORTE           Emolor Vocals      METRO    IV  
1        METRO-NORTE           Emolor Vocals      METRO    IV  
2        METRO-NORTE           Emolor Vocals      METRO    IV  
3        METRO-NORTE           Emolor Vocals      METRO    IV  
4        METRO-NORTE           Emolor Vocals      METRO    IV  
...              ...                     ...        ...   ...  
4862344      NORESTE       Thrilmamad Vocals    FORANEO    II  
4862345      NORESTE       Thrilmamad Vocals    FORANEO    II  
4862346      NORESTE       Thrilmamad Vocals    FORANEO    II  
4862347      NORESTE       Thrilmamad Vocals    FORANEO    II  
4862348      NORESTE       Thrilmamad Vocals    FORANEO    II  

[6039729 rows x 14 columns]

In [53]:
# Checking the concat was done correctly
df_merge1.shape[0]+df_merge2.shape[0]

6039729

In [54]:
sellin2 = sellin.head()
sellin2

Sku       Date  Store_Id  Pieces  Price  Margin
0  4220015 2015-02-01       100    20.0  13.01    9.78
1  4220015 2015-02-01       116    20.0  13.01    9.78
2  4220015 2015-02-01       141    80.0  13.01    9.78
3  4220015 2015-02-01       146    10.0  13.01    9.78
4  4220015 2015-02-01       151    20.0  13.09    9.87

In [55]:
x = True
acum=0

while x == True:
    sellin2 = sellin.head(acum+1)
    df_merge = sellin2.merge(maestro_client)
    if len(df_merge)== len(sellin2):
        acum = len(sellin2)
        print(acum)
    else:
        print("STOP", len(sellin2))
        x = False

1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
178
179
180
181
182
183
184
185
186
187
188
189
190
191
192
193
194
195
196
197
198
199
200
201
202
203
204
205
206
207
208
209
210
211
212
213
214
215
216
217
218
219
220
221
222
223
224
225
226
227
228
229
230
231
232
233
234
235
236
237
238
239
240
241
242
243
244
245
246
247
248
249
250
251
252
253
254
255
256
257
258
259
260
261
262
263
264
265
266
267
268
269
270
271
272
273
274
275
276
277


In [56]:
sellin3 = sellin.iloc[810:820, :]
df_merge3 =  sellin3.merge(maestro_client)
df_merge3

Sku       Date  Store_Id  Pieces  Price  Margin  \
0   4220015 2015-08-01      5817    20.0  12.59    9.36   
1   4220015 2015-09-01       100    40.0  13.01    9.78   
2   4220015 2015-09-01       121    20.0  13.01    9.78   
3   4220015 2015-09-01       129    40.0  13.01    9.78   
4   4220015 2015-09-01       231    20.0  13.01    9.78   
5   4220015 2015-09-01       233    40.0  13.01    9.78   
6   4220015 2015-09-01       238    60.0  13.09    9.87   
7   4220015 2015-09-01       239    40.0  13.01    9.78   
8   4220015 2015-09-01       240    20.0  13.01    9.78   
9   4220015 2015-09-01       240    20.0  13.01    9.78   
10  4220015 2015-09-01       260    40.0  13.01    9.78   

               NOMBRE DE TDA       CADENA     FORMATO  \
0   Awesome Yellow Stationer  proveedor_2  GipsyTrade   
1        Clean Grey Jeweller  proveedor_1        Shop   
2    Massive Purple HighShop  proveedor_1        Shop   
3   Dirty Lavender SmallShop  proveedor_1        Shop   
4        Awesome Pink BigCom  proveedor_1        Shop   
5      Defiant Silver Center  proveedor_1        Shop   
6    Bloody Lime Convenience  proveedor_1        Shop   
7           Dirty Lemon Shop  proveedor_1        Shop   
8         Dull Navy BookShop  proveedor_1        Shop   
9         Dull Navy BookShop  proveedor_2    TinyShop   
10      Tiny Lemon GreatShop  proveedor_1        Shop   

           MUNICIPIO/CIUDAD     ESTADO     DIVISION  \
0   Enlightened Dark Empire  Magornmar    METRO-SUR   
1          Funny Lemon Star  Magornmar    METRO-SUR   
2          Funny Lemon Star  Magornmar    METRO-SUR   
3          Funny Lemon Star  Magornmar    METRO-SUR   
4    Interesting Green Pine   Alloralf    METRO-SUR   
5          Funny Lemon Star  Magornmar    METRO-SUR   
6           Clean Aqua Soil    Egusgul   CENTRO-SUR   
7      Young Orchid Capital  Magornmar  METRO-NORTE   
8            Rare Lemon Key   Alloralf    METRO-SUR   
9            Rare Lemon Key   Alloralf  METRO-NORTE   
10           Rare Lemon Key   Alloralf  METRO-NORTE   

         DEPOSITO / DISTRIBUIDOR ESTRUCTURA GRADO  
0          Thrilvitar Industrail      METRO     V  
1              Thrilthoal Empire      METRO     V  
2              Thrilthoal Empire      METRO     V  
3   Ioalfio Ifonforge Industries      METRO     V  
4   Ioalfio Ifonforge Industries      METRO    IV  
5              Thrilthoal Empire      METRO     V  
6             Erodmad Industrail    FORANEO    IV  
7                  Emolor Vocals      METRO     V  
8              Thrilthoal Empire      METRO    IV  
9           Lorthogus Toughworks      METRO    IV  
10             Thrilvallor Corp.      METRO    IV